### PR TITLE
Pagination tests

### PR DESCRIPTION
**Pull request by RickyV33**
Adds unit tests for the pagination module

### CODE REVIEW

File: test/lib/pagination.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        });
+      };
+      pagination = proxyquire('../../lib/pagination', { 'request': requestStub });
+      expect(pagination('http://notdummy:password@sevensource.jamacloud.com/rest/latest/projects', startAt, maxResults)).to.eventually.be.rejected();
+    });
+    //  Test #2: password is invalid
+    //  returned promise from pagination should be rejected
+    it('should reject if the password is invalid', function () {
+      setStartAtAndMaxResults(0, 20);
+      let requestStub = function (options, callback) {
```
Comment (RickyV33) [inline test/lib/pagination.js]: requestStub seems to be reused a bunch throughout the test. You should pull it out so you can just have one declaration

**Comment by ibilal** [comment]
Hey all, so the last test is failing. It only seems to be returning an array of size 20. If someone could take a look at it. Thanks!

File: lib/pagination.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ function pagination (baseUrl, startAt, maxResults) {
       if (resultCount === 0) {
         resolve(payload);
       } else {
-        let maxResultsParam = 'maxResults=' + PAGE_SIZE;
+        let maxResultsParam = 'maxResultsAllowed=' + PAGE_SIZE;
```
Comment (RickyV33) [inline lib/pagination.js]: This should be maxResults. I tried it with postman and if I want only 3 results, it still returns the whole thing with 'maxResultsAllowed'

**Comment by ibilal** [comment]
Test 11 is now test 9 and is good to go.

**Comment by chances** [comment]
Minor nitpick:

```javascript
let callBackObject = pages[page].body;
callback(null, {body: callBackObject});
```
could be simplified in the five occurrences to:
```javascript
callback(null, pages[page]);
```

Apart from that, LGTM!

File: lib/pagination.js (hunk 2 above)
Comment (RickyV33) [inline lib/pagination.js]: @ibilal This still needs to be changed before it can be pulled in

**Pull request merged**